app/graphql/resolvers/corpus_slate_lineup_resolvers.py

```diff
@@ -46,14 +48,19 @@ async def resolve_home_slate_lineup(root, info: Info) -> CorpusSlateLineup:
         topic_provider=topic_provider,
         for_you_slate_provider=ForYouSlateProvider(corpus_client),
         recommended_reads_slate_provider=RecommendedReadsSlateProvider(corpus_client),
-        topic_slate_provider=TopicSlateProvider(corpus_client),
+        topic_slate_providers=TopicSlateProviderFactory(corpus_client),
         collection_slate_provider=CollectionSlateProvider(corpus_client),
     ).get_slate_lineup(
         user=user,
         slate_count=slate_count,
         recommendation_count=recommendation_count,
     )
 
+    slate_lineup_tracker = SnowplowCorpusSlateLineupTracker(
```

Comment:
For your performance work later this week. It seems like we are constructing a lot of session and provider classes in this resolver. I know that boto3 clients in particular can be expensive to build. I would bet we could get some performance improvements by moving these items to statics or other factories where they are created only once. It would be even better if we could inject all of these providers into this function.

Reply:
Agreed. That's on my radar as well.